state/protocol/inmem/epoch.go

```diff
@@ -155,6 +156,7 @@ func (es *committedEpoch) Cluster(index uint) (protocol.Cluster, error) {
 		return nil, fmt.Errorf("failed to generate clustering: %w", err)
 	}
 
+	// TODO: double check ByIndex returns canonical order
 	members, ok := clustering.ByIndex(index)
```

Comment:
⚠️ Currently it respects the assignment order in the `EpochSetup` event, which is not the same as the canonical ordering used by eg. `Snapshot.Identities`. (see previous comment)

Reply:
Yes, will be addressed in separate PR